llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -18,32 +18,29 @@ using namespace llvm;
 
 void llvm::getSYCLDeviceRequirements(
     const Module &M, std::map<StringRef, std::vector<uint32_t>> &Requirements) {
-  auto ExtractIntegerFromMDNodeOperand = [=](const MDNode *N,
-                                             unsigned OpNo) -> unsigned {
-    Constant *C =
-        cast<ConstantAsMetadata>(N->getOperand(OpNo).get())->getValue();
-    return static_cast<uint32_t>(C->getUniqueInteger().getZExtValue());
+  std::set<uint32_t> Aspects;
+  auto FindAspectsByMDName = [&](const Function &F, std::string MDName) {
+    if (const MDNode *MDN = F.getMetadata(MDName))
+      for (size_t I = 0, E = MDN->getNumOperands(); I < E; ++I) {
+        Constant *C =
+            cast<ConstantAsMetadata>(MDN->getOperand(I).get())->getValue();
+        Aspects.insert(
+            static_cast<uint32_t>(C->getUniqueInteger().getZExtValue()));
+      }
   };
 
-  // { LLVM-IR metadata name , [SYCL/Device requirements] property name }, see:
-  // https://github.com/intel/llvm/blob/sycl/sycl/doc/design/OptionalDeviceFeatures.md#create-the-sycldevice-requirements-property-set
-  // Scan the module and if the metadata is present fill the corresponing
-  // property with metadata's aspects
```

Comment:
I feel like we are losing information by removing this. Would it make sense to keep this comment, potentially rephrased?

Reply:
I agree, it's mostly the second part that still seems relevant.

Reply:
Done in 22fd515